presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkPhysicalResourceCalculator.java

```diff
@@ -0,0 +1,148 @@
+            isCurrentQueryHourPlus = (inputDataInGB.compareTo(AVERAGE_SIZE_FOR_SHORTER_QUERIES) > 0) ? true : false;
+        }
+        else if (queryTypeByRunTime.equals(QueryTypeByRunTime.LONG)) {
+            isCurrentQueryHourPlus = true;
+        }
+
+        this.executorCount = calculateExecutorCount(session, inputDataInGB, isCurrentQueryHourPlus);
+        this.shufflePartitionCount = calculateShufflePartitionCount(session, executorCount, inputDataInGB, isCurrentQueryHourPlus);
+
+        return new PhysicalResourceSettings(executorCount, shufflePartitionCount);
+    }
+
+    private QueryTypeByRunTime getQueryType(Duration targetedRuntimeForQuery)
+    {
+        if (targetedRuntimeForQuery.compareTo(Duration.valueOf("1h")) > 0) {
+            return QueryTypeByRunTime.LONG;
+        }
+        else if (targetedRuntimeForQuery.equals(Duration.valueOf("0m"))) {
+            return QueryTypeByRunTime.NOT_DEFINED;
+        }
+        else {
+            return QueryTypeByRunTime.SMALL;
+        }
+    }
+
+    private int calculateExecutorCount(Session session, DataSize inputDataInGB, boolean isCurrentQueryHourPlus)
+    {
+        double resourceUsagePerExecutor = getResourceUsagePerExecutor(session).getValue();
+        double calculatedNumberOfExecutors = inputDataInGB.getValue() / resourceUsagePerExecutor;
+        int minExecutorCount = getMinExecutorCountForResourceAllocation(session);
+        int maxExecutorCount = getMaxExecutorForResourceAllocation(session);
+        if (false == isCurrentQueryHourPlus) {
+            minExecutorCount = (int) (minExecutorCount * SCALING_FACTOR_FOR_EXECUTOR_COUNT_BASED_ON_RUNTIME);
+            maxExecutorCount = (int) (maxExecutorCount * SCALING_FACTOR_FOR_EXECUTOR_COUNT_BASED_ON_RUNTIME);
+        }
+        int updatedExecutorCount = Math.max((int) calculatedNumberOfExecutors, minExecutorCount);
+        int tunedExecutorCount = Math.max(minExecutorCount, Math.min(maxExecutorCount, updatedExecutorCount));
+
+        return tunedExecutorCount;
+    }
+
+    private int calculateShufflePartitionCount(Session session, int executorCount, DataSize inputDataInGB, boolean isCurrentQueryHourPlus)
+    {
+        int maxShufflePartitionCount = getMaxShufflePartitionCount(session);
+        int minShufflePartitionCount = getMinShufflePartitionCount(session);
+
+        double resourceUsagePerPartitionCount = getResourceUsagePerPartitionCount(session).getValue();
+        double calculatedNumberOfPartitionCount = inputDataInGB.getValue() / resourceUsagePerPartitionCount;
+
+        int lowerBound = Math.abs(minShufflePartitionCount - (int) calculatedNumberOfPartitionCount);
+        int upperBound = Math.abs(maxShufflePartitionCount - (int) calculatedNumberOfPartitionCount);
+
+        int tunedShufflePartitionCount = ((lowerBound - upperBound) > 0) ? maxShufflePartitionCount : minShufflePartitionCount;
+
+        return tunedShufflePartitionCount;
```

Comment:
Correct - have never tested with continuous partition count allocation. I am choosing the min or max based on the closeness of the calculated partition count. Since @pgupta2 you mentioned that spark is using continuous partition count moving it to the same model.